SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/CodeGenerator/Unreal/SourceGenerator.cs

```diff
@@ -0,0 +1,352 @@
+{{
+{Text.Indent(1, type.Fields.Count == 0 ? "return true;"
+: $"return {string.Join($@" && {Environment.NewLine}", type.Fields.Select(f => Types.GetFieldDefinitionEquals(f, $"_{Text.SnakeCaseToPascalCase(f.Identifier.Name)}", "Value")))};")}
+}}
+
+bool {name}::Update::operator!=(const {name}::Update& Value) const
+{{
+{Text.Indent(1, $"return !operator== (Value);")}
+}}
+
+{name}::Update {name}::Update::FromInitialData(const {name}& Data)
+{{ 
+{Text.Indent(1, $@"{name}::Update Update;
+{string.Join(Environment.NewLine, type.Fields.Select(f => $"Update._{Text.SnakeCaseToPascalCase(f.Identifier.Name)} = Data.Get{Text.SnakeCaseToPascalCase(f.Identifier.Name)}();"))}
+return Update;")}
+}}
+
+{name} {name}::Update::ToInitialData() const
+{{
+{Text.Indent(1, $@"return {name} (
+{string.Join($",{Environment.NewLine}", type.Fields.Select(f => Text.Indent(1, $"*_{Text.SnakeCaseToPascalCase(f.Identifier.Name)}")))});")}
+}}         
+
+void {name}::Update::ApplyTo({name}& Data) const
+{{
+{Text.Indent(1, string.Join(Environment.NewLine, type.Fields.Select(field => $@"if (_{Text.SnakeCaseToPascalCase(field.Identifier.Name)})
+{{
+{Text.Indent(1, $@"Data.Set{Text.SnakeCaseToPascalCase(field.Identifier.Name)}(*_{Text.SnakeCaseToPascalCase(field.Identifier.Name)});")}
+}}")))}
+}}
+");
+            if (type.Fields.Count() > 0)
+            {
+                builder.AppendLine(string.Join(Environment.NewLine, type.Fields.Select(field => $@"const {Types.CollectionTypesToQualifiedTypes[Types.Collection.Option]}<{Types.GetFieldTypeAsCpp(field, bundle, type)}>& {name}::Update::Get{Text.SnakeCaseToPascalCase(field.Identifier.Name)}() const
+{{
+{Text.Indent(1, $"return _{Text.SnakeCaseToPascalCase(field.Identifier.Name)};")}
+}}
+
+{Types.CollectionTypesToQualifiedTypes[Types.Collection.Option]}<{Types.GetFieldTypeAsCpp(field, bundle, type)}>& {name}::Update::Get{Text.SnakeCaseToPascalCase(field.Identifier.Name)}()
+{{
+{Text.Indent(1, $"return _{Text.SnakeCaseToPascalCase(field.Identifier.Name)};")}
+}}
+
+{name}::Update& {name}::Update::Set{Text.SnakeCaseToPascalCase(field.Identifier.Name)}({Types.GetConstAccessorTypeModification(field, bundle, type)} value)
+{{
+{Text.Indent(1, $@"_{ Text.SnakeCaseToPascalCase(field.Identifier.Name)} = value;
+return *this;")}
+}}
+")));
+            }
+
+            if (type.Events.Count() > 0)
+            {
+                builder.AppendLine(string.Join(Environment.NewLine, type.Events.Select(_event => $@"const {Types.CollectionTypesToQualifiedTypes[Types.Collection.List]}< {Types.GetTypeDisplayName(_event.Type.Type.QualifiedName)} >& {name}::Update::Get{Text.SnakeCaseToPascalCase(_event.Identifier.Name)}List() const
+{{
+{Text.Indent(1, $"return _{Text.SnakeCaseToPascalCase(_event.Identifier.Name)}List;")}
+}}
+
+{Types.CollectionTypesToQualifiedTypes[Types.Collection.List]}< {Types.GetTypeDisplayName(_event.Type.Type.QualifiedName)} >& {name}::Update::Get({Text.SnakeCaseToPascalCase(_event.Identifier.Name)}List()
```

Comment:
```suggestion
{Types.CollectionTypesToQualifiedTypes[Types.Collection.List]}< {Types.GetTypeDisplayName(_event.Type.Type.QualifiedName)} >& {name}::Update::Get({Text.SnakeCaseToPascalCase(_event.Identifier.Name)}List())
```

We're missing a bracket in the end I think.